### PR TITLE
chore: Add proper deprecation dates where missing

### PR DESCRIPTION
**Pull request by susnux**
## Summary

Looked up deprecation date where missing.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

File: lib/public/Share_Backend.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ public function isValidSource($itemSource, $uidOwner);
 	 * Get a unique name of the item for the specified user
 	 * @param string $itemSource
 	 * @param string|false $shareWith User the item is being shared with
-	 * @param array|null $exclude List of similar item names already existing as shared items @deprecated since version OC7
+	 * @param array|null $exclude List of similar item names already existing as shared items @deprecated 7.0.0
```
Comment (come-nc) [inline lib/public/Share_Backend.php:33]:
```suggestion
	 * @param array|null $exclude List of similar item names already existing as shared items 
	 * @deprecated 7.0.0
```

**Review by come-nc — APPROVED** [review]

**Review by provokateurin — APPROVED** [review]
Can you also extend the regex to enforce this?
And please also backport what is possible.

File: lib/public/Share_Backend.php (hunk 1 above)
Comment (susnux) [reply lib/public/Share_Backend.php:33]: I think only the property is deprecated?

Comment (provokateurin) [reply lib/public/Share_Backend.php:33]: That's not really possible. And having two annotations in the same line doesn't work syntax wise

Comment (come-nc) [reply lib/public/Share_Backend.php:33]: It’s never used in any application I have locally, nor in server.

Comment (come-nc) [reply lib/public/Share_Backend.php:33]: ping @icewind1991 @nickvergessen What do you think, was the whole method deprecated in OC7 or only the parameter? What do we do with it?

Comment (susnux) [reply lib/public/Share_Backend.php:33]: Original comment author was @nickvergessen in https://github.com/nextcloud/server/commit/dfde04291eae31765b2cb858e1d1260b60d75329

**Pull request merged**